pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2555,36 +2555,30 @@ private boolean isSystemTopic(String topic) {
     }
 
     /**
-     * Get {@link TopicPolicies} for this topic.
+     * Get {@link TopicPolicies} for the parameterized topic.
      * @param topicName
      * @return TopicPolicies is exist else return null.
      */
     public TopicPolicies getTopicPolicies(TopicName topicName) {
+        if (!pulsar().getConfig().isTopicLevelPoliciesEnabled()) {
+            return null;
+        }
         TopicName cloneTopicName = topicName;
         if (topicName.isPartitioned()) {
             cloneTopicName = TopicName.get(topicName.getPartitionedTopicName());
         }
         try {
-            checkTopicLevelPolicyEnable();
             return pulsar.getTopicPoliciesService().getTopicPolicies(cloneTopicName);
         } catch (BrokerServiceException.TopicPoliciesCacheNotInitException e) {
             log.debug("Topic {} policies have not been initialized yet.", topicName.getPartitionedTopicName());
             return null;
-        } catch (RestException | NullPointerException e) {
+        } catch (NullPointerException e) {
```

Comment:
Get we get rid of this `catch NPE`.
It is really a code smell.
I see it was only moved from another point, but it is a good time to fix things when we put our hands on.

Reply:
I agree that this is code smell. I don't see any way for this `NPE` to happen (even before this PR's changes), and now that we're checking the configuration before getting the topic policy, I see no need for it. I'll remove it.